xarray/core/missing.py

```diff
@@ -755,8 +761,9 @@ def interp_func(var, x, new_x, method: InterpOptions, kwargs):
 
 def _interp1d(var, x, new_x, func, kwargs):
     # x, new_x are tuples of size 1.
-    x, new_x = x[0], new_x[0]
-    rslt = func(x, var, assume_sorted=True, **kwargs)(np.ravel(new_x))
+    x, new_x = x[0].data, new_x[0].data
```

Comment:
Is there a reason you got rid of `assume_sorted=True`

Reply:
1. This doesn't seem to have been necessary, changing back

2. The interpolators I have added (`Scipy.Interpolate.Akima1DInterpolator`, etc...) require sorted axes but don't take `assume_sorted` as an argument. Only `Scipy.Interpolate.Interp1d` takes this argument. Since we are always calling it with `True`,  I'm just letting the default for ScipyInterpolator init impose this. 

There is maybe a philosophical conflict here and in previous comments about kwarg handling for the different interpolators. The preexisting script (`missing.py`) has special classes (`NumpyInterpolator`, `ScipyInterpolator`...) which basically just do this handling. I didn't add any new classes for the new interpolators and rather tried to do everything with kwargs determined by `_get_interpolator`.

Reply:
Re (1): If you're comfortable with typing, some type hints would go a long way towards improving this module. 

Re (2): sounds good

Reply:
(1) Yeah can do! Prefer a separate commit though (?)

Reply:
we squash merge so it doesn't matter :) unless you meant separate PR, in which case that would be fine too.